spec/ParseGraphQLServer.spec.js

```diff
@@ -9515,6 +9515,118 @@ describe('ParseGraphQLServer', () => {
           }
         });
 
+        it('should support where argument on object field that contains boolean value', async () => {
+          try {
+            const someObjectFieldValue = {
+              foo: { bar: 'baz', qux: true },
+              number: 10,
+            };
+
+            await apolloClient.mutate({
+              mutation: gql`
+                mutation CreateClass($schemaFields: SchemaFieldsInput) {
+                  createClass(input: { name: "SomeClass", schemaFields: $schemaFields }) {
+                    clientMutationId
+                  }
+                }
+              `,
+              variables: {
+                schemaFields: {
+                  addObjects: [{ name: 'someObjectField' }],
+                },
+              },
+              context: {
+                headers: {
+                  'X-Parse-Master-Key': 'test',
+                },
+              },
+            });
+            await parseGraphQLServer.parseGraphQLSchema.schemaCache.clear();
+
+            const schema = await new Parse.Schema('SomeClass').get();
+            expect(schema.fields.someObjectField.type).toEqual('Object');
+
+            const createResult = await apolloClient.mutate({
+              mutation: gql`
+                mutation CreateSomeObject($fields: CreateSomeClassFieldsInput) {
+                  createSomeClass(input: { fields: $fields }) {
+                    someClass {
+                      id
+                    }
+                  }
+                }
+              `,
+              variables: {
+                fields: {
+                  someObjectField: someObjectFieldValue,
+                },
+              },
+            });
+
+            const whereWithQuxFalse = {
+              someObjectField: {
+                notEqualTo: { key: 'foo.bar', value: 'bat' },
+                greaterThan: { key: 'number', value: 9 },
+                lessThan: { key: 'number', value: 11 },
+                equalTo: { key: 'foo.qux', value: false },
+              },
+            };
+            const whereWithQuxTrue = {
+              someObjectField: {
+                ...whereWithQuxFalse.someObjectField,
+                equalTo: { key: 'foo.qux', value: true },
+              },
+            };
+            const queryResult = await apolloClient.query({
+              query: gql`
+                query GetSomeObject(
+                  $id: ID!
+                  $whereWithQuxFalse: SomeClassWhereInput
+                  $whereWithQuxTrue: SomeClassWhereInput
+                ) {
+                  someClass(id: $id) {
+                    id
+                    someObjectField
+                  }
+                  someClasses(where: $whereWithQuxFalse) {
+                    edges {
+                      node {
+                        id
+                        someObjectField
```

Comment:
@yomybaby  i can suggest here also to add a query on a field of type number === 0. Since your PR also fix the issue if a developer query an object field with `value: 0`